gistey/auth.py

```diff
@@ -0,0 +1,47 @@
+import requests
+import datetime
+import time
+import getpass
+from functools import partial
+
+end = "https://api.github.com/"
+
+
+def authorize(func):
+    """
+    Authorize application
+
+    :param func:
+        function to decorate for required authorization.
+    """
+
+    username = input('Enter GitHub Username: ')
+    password = getpass.getpass()
+    return partial(func, auth=(username, password))
+
+
+@authorize
```

Comment:
There seems to be a problem in making `authorize` function as the decorator. 
Because currently, the decorator right away asks for `username` and `password` instead of asking at function call.